lambda/matching.ml

```diff
@@ -2319,6 +2348,7 @@ let get_expr_args_array ~scopes kind head arg rem =
       {
         arg;
         binding_kind = StrictOpt;
+        mut = compose_mut mut Mutable;
```

Comment:
I am not sure how I feel having constant multiplications by `0` or `1` in the source code, but why not.

Reply:
With this code organization I find it much easier to check how mutability information evolves during compilation. (Also: if I wanted to track both the "transitive mutability" and the "immediate mutability" of the argument position, this would be easy to do by changing `mut` here to be a pair, without changing much of the rest of the code. This would let us get rid of `mut_of_binding_kind`. I decided not to do it because there is bigger fish to fry, but it suggests that `compose_mut` is a nice abstraction to have.)